### PR TITLE
Refactored docker-compose.yml

### LINKED ISSUE #73: Remove duplicated code in docker-compose.yml
The docker-compose.yml has several parts of duplicated code, we need remove the dupicated code, this can help https://docs.docker.com/compose/compose-file/#extension-fields.

### PR DESCRIPTION
**Pull request by sudiptog81**
#### Changes
Refactored duplicate sections in `docker-compose.yml` to configuration fragments using extension fields.

Fixes and closes #73 .

#### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by riibeirogabriel — COMMENTED** [review]
Hi @sudiptog81, please take a look from my comment and thanks to make this issue!

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,18 @@
-version: '3'
+version: "3.8"
+
+x-deploy: &default-deploy
+    replicas: 1
+    restart_policy:
+        condition: on-failure
+    placement:
+        constraints:
+            - "node.role==manager"
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply docker-compose.yml:9]: Sure, taking care in the next commit.

**Comment by sudiptog81** [comment]
Per the newer `docker-compose.yml`, the `diff` between the output of `docker-compose -f docker-compose.yml config` is the following, mostly due to config version change from `3` to `3.8` which allows extension fields.

```diff
25c25,26
<     - 5000:5000/tcp
---
>     - published: 5000
>       target: 5000
67c68,69
<     - 27017:27017/tcp
---
>     - published: 27017
>       target: 27017
113c115,116
<     - 5003:5003/tcp
---
>     - published: 5003
>       target: 5003
136c139,140
<     - 5004:5004/tcp
---
>     - published: 5004
>       target: 5004
149c153,154
<     - 5050:5000/tcp
---
>     - published: 5050
>       target: 5000
179,180c184,187
<     - 5002:5002/tcp
<     - 41100:41100/tcp
---
>     - published: 5002
>       target: 5002
>     - published: 41100
>       target: 41100
208,209c215,218
<     - 5006:5006/tcp
<     - 41300:41300/tcp
---
>     - published: 5006
>       target: 5006
>     - published: 41300
>       target: 41300
239,240c248,251
<     - 5001:5001/tcp
<     - 41000:41000/tcp
---
>     - published: 5001
>       target: 5001
>     - published: 41000
>       target: 41000
258,260c269,274
<     - 4040:4040/tcp
<     - 7077:7077/tcp
<     - 8080:8080/tcp
---
>     - published: 4040
>       target: 4040
>     - published: 7077
>       target: 7077
>     - published: 8080
>       target: 8080
280,281c294,297
<     - 8081:8081/tcp
<     - 41352:41352/tcp
---
>     - published: 8081
>       target: 8081
>     - published: 41352
>       target: 41352
309,310c325,328
<     - 5005:5005/tcp
<     - 41200:41200/tcp
---
>     - published: 5005
>       target: 5005
>     - published: 41200
>       target: 41200
323c341,342
<     - 80:8080/tcp
---
>     - published: 80
>       target: 8080
326c345
< version: '3.0'
---
> version: '3.8'
```

File: docker-compose.yml [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+
+x-deploy-worker: &default-deploy-worker
+    <<: *default-deploy
+    placement:
+        constraints:
+            - "node.role==worker"
+
+x-class-env: &default-class-env
```
Comment (riibeirogabriel) [inline docker-compose.yml:20]: A more descriptive name to this like "database-env" sounds better?

**Review by riibeirogabriel — COMMENTED** [review]
Please take a look in my comment

File: docker-compose.yml (hunk 2 above)
Comment (riibeirogabriel) [reply docker-compose.yml:20]: You make a great work, thanks!

Comment (sudiptog81) [reply docker-compose.yml:20]: I was even wondering if we can refactor the database environment to remove duplicacies between `databaseprimary`, `databasesecondary` and `databasearbiter`. However there are some variables which are different so I left those untouched. I feel we can rename `x-class-env` and `default-class-env` to `x-service-database-env` and `default-service-database-env` respectively?

Comment (riibeirogabriel) [reply docker-compose.yml]: Sounds good!

**Review by riibeirogabriel — COMMENTED** [review]
I wil test this changes in production and then i will merge if everything happens good, okay?

**Comment by riibeirogabriel** [comment]
@sudiptog81 Do you have interest in continue contributing with this project? is is the first release yet, more features are coming, and please give a star to help the project, and thanks!

**Comment by sudiptog81** [comment]
> I wil test this changes in production and then i will merge if everything happens good, okay?

Sure! 😃 



> @sudiptog81 Do you have interest in continue contributing with this project? is is the first release yet, more features are coming, and please give a star to help the project, and thanks!

Would love to, have some experience working with microservices, and this is an awesome project to collaborate on.

**Review by riibeirogabriel — APPROVED** [review]

**Pull request merged**